### ISSUE TITLE
Invalid HTTP_HOST header when application and PostHog in same docker network

### ISSUE THREAD

**Issue by GlennHyland** [PostHog/posthog#1734]
When we have a Django application container in the same docker network as our PostHog deployment, we get the following error :
`posthog_web_1     | Invalid HTTP_HOST header: 'posthog_web:8000'. The domain name provided is not valid according to RFC 1034/1035.`

In Django the apps.py has the host set to below :
`        posthog.host = 'http://posthog_web:8000'
`

Adding a 'Host' entry in the headers variable in post function in posthog/request.py, and using just the hostname (no port) for that value fixes the issue :
```
    import urllib
    parse_host_url = urllib.parse.urlparse(host)
    hostname = parse_host_url.netloc.split(":")[0]
    headers = {
        'Content-Type': 'application/json',
        'User-Agent': 'analytics-python/' + VERSION,
        'Host': hostname
    }
```

**Comment by timgl** [comment]
Hey! I'm struggling to reproduce this locally in a test, but it looks like you've already got a solution to this? If you'd like, feel free to raise a PR to fix this here: https://github.com/posthog/posthog-python

Thanks!

**Comment by GlennHyland** [comment]
Hey,
It looks like I jumped the gun here. The issue is with my configuration, the service name in docker-compose I use for posthog is posthog_web, and had not defined a hostname. The "_" is not allowed in the HOST name as per RFC 1034/1035 and that was what was causing the Invalid HTTP_HOST header. 
I updated my docker compose to set hostname :
`   hostname: posthog-web`
and my django apps.py using the same :
`posthog.host = 'http://posthog-web:8000'`

Now there are no errors and Django can send the metrics to the posthog server.

Apologies for any time wasted.
Thanks!
